static/app/components/nav/utils.tsx

```diff
@@ -97,17 +119,23 @@ export function isSubmenuItemActive(
   );
 }
 
-/** Creates a `LocationDescriptor` from a URL string that may contain search params */
+/**
+ ** Creates a `LocationDescriptor` from a URL string that may contain search params
+ */
 export function makeLinkPropsFromTo(to: string): {
   state: object;
   to: LocationDescriptor;
 } {
-  const [pathname, search] = to.split('?');
+  const {pathname, search, hash} = new URL(
+    to,
+    !to.startsWith('http') ? 'https://sentry.io/' : undefined
```

Comment:
Should we make this a constant somewhere? It looks like something that we'd want to be able to easily change

Reply:
I can use a constant, but this value is never actually used. `URL` expects a fully formed URL string, so we need to pass something in as a base. `http://example.com/` would also work here.

Reply:
hmmm, does this work ok for on premise and things like that?

Reply:
I added a comment above this code to explain more! 

Also made sure to pass this through `normalizeUrl()` to address on-prem concerns. See 64ab49c1d2c.